apps/publisher/extensions/assets/default/apis/assets.jag

```diff
@@ -68,34 +68,45 @@ require('/modules/publisher.js').exec(function (ctx) {
             var newAsset;
             if (uriMatcher.match(CREATE_URL)) {//to create asset
                 try {
-                    newAsset = assetApi.create(options, req, res, session);
-                    //If the asset is successfully created then return the asset
-                    if (newAsset) {
-                        response = responseProcessor.buildSuccessResponse(res,
-                                constants.STATUS_CODES.CREATED, newAsset)
+                    var contentTypeheader = req.getHeader('Content-Type');
+                    var n = contentTypeheader.indexOf("multipart/form-data");
+                    if (contentTypeheader === 'application/json' || contentTypeheader === 'application/x-www-form-urlencoded' || n >= 0) {
+                        newAsset = assetApi.create(options, req, res, session);
+                        var requestURI = req.getRequestURI();
+                        //If the asset is successfully created then return the asset
+                        if (newAsset) {
+                            response = responseProcessor.buildSuccessResponse(res, constants.STATUS_CODES.CREATED, newAsset);
+                            response.addHeader('Location', requestURI + "/" + newAsset.id + "?type=" + newAsset.type);
+                        } else {
+                            //response = responseProcessor.buildErrorResponse(res,constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'Failed to create asset of type: ' +options.type);
+                            response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'error on create asset', response, 'Failed to create asset of type: ' + options.type, "more information on create asset error", []);
+                        }
                     } else {
-                        response = responseProcessor.buildErrorResponse(res,
-                                constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'Failed to create asset of type: ' +
-                                        options.type);
+                        response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.UNSUPPORTED_MEDIATYPE, 'error on create asset', response, 'Unsupported Media Type of creating ' + options.type, "more information on create asset error", []);
                     }
                 } catch (e) {
                     log.error(e);
```

Comment:
As now we have log.error(msg, exception) support, please refactor this to use log.error(e.getMessage(), e);